app/helpers/discovered_hosts_helper.rb

```diff
@@ -81,4 +81,15 @@ def discovery_status_icon(host)
     "<span class='glyphicon #{status_glyph}' style='color: #{status_color}'
       title='#{status_message}'/>".html_safe
   end
+
+  def self.normalize_string_for_hostname(hostname)
+    hostname.to_s.downcase.gsub(/(^[^a-z0-9]|[^a-z0-9-_]|[^a-z0-9]$)/,'')
+  end
+
+  def self.return_first_valid_fact(facts_array,facts)
+    return if !facts_array.is_a?(Array)
+    facts_array.each do |value|
+      return facts[value] if !facts[value].nil?
+    end
+  end
```

Comment:
@lzap looking at this again, I am not sure these methods belongs here. I am thinking of just moving them into the model, as class methods. What do you think?